torch/__init__.py

```diff
@@ -176,6 +176,13 @@ def set_default_dtype(d):
     """
     _C._set_default_dtype(d)
 
+
+def set_memory_format_propagation(b):
+    _C._set_memory_format_propagation(b)
+
+def get_memory_format_propagation():
+    return _C._get_memory_format_propagation()
```

Comment:
I want to bikeshed the API here. I'll give justification for the bikeshedding I suggest.

First, let's move this out of `torch` top-level namespace, and into a module like `torch.__future__`. Because the intent of this is to NOT be a permanent feature toggle, but rather a way to opt into what will (eventually) be the default behavior, it needs to be extremely clear to end users that this is NOT a supported way to toggle between one mode or another, it is purely a BC knob. In Python, there is an existing convention for this: `from __future__ import absolute_import`, for example. I believe that using a name similar to `__future__` will inform people about the intent appropriately.

Second, let's not have a getter/setter function, but instead have this be a regular variable living in the namespace, e.g., `torch.__future__.propagate_memory_format = True`. This brings this toggle in line with other similar global toggles we have, such as `torch.backends.cudnn.enabled = True`.

I also have some deeper points about the semantics of this flag. More on this shortly.